File: front/src/app/Models/Document.ts
import { Bacteriology } from "./Bacteriology";
import { Biology } from "./Biology";
import { BloodGroup } from "./BloodGroup";
import { Radiology } from "./Radiology";
import { Serology } from "./Serology";
import { SpermAnalysis } from "./SpermAnalysis";

export class Document {
    biologies!: Biology[];
    bacteriologies!: Bacteriology[];
    bloodgroups!: BloodGroup[];
    radiologies!: Radiology[];
    serologies!: Serology[];
    spermAnalyses!: SpermAnalysis[];
}

File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/AnalyseDTOs/Document.java
package org.example.medicalreport.Models.AnalyseDTOs;

import lombok.Data;

import java.util.List;

@Data
public class Document {
    List<Biology> biologies;
    List<Bacteriology> bacteriologies;
    List<BloodGroup> bloodgroups;
    List<Radiology> radiologies;
    List<Serology> serologies;
    List<SpermAnalysis> spermAnalyses;
}
